packages/xo-web/src/common/sorted-table/index.js

```diff
@@ -704,6 +631,65 @@ export default class SortedTable extends Component {
     this._selectItem(+target.name, target.checked, event.nativeEvent.shiftKey)
   }
 
+  _getParsedQueryString = createSelector(
+    () => this.props.router.location.query[this.props.stateUrlParam],
+    urlState => {
+      let matches
+      const [, page, selectedColumnId, sortOrder, filter] =
+        urlState !== undefined &&
+        (matches = URL_STATE_RE.exec(urlState)) !== null
+          ? matches
+          : []
```

Comment:
If `urlState` is `undefined`, the filter will take `undefined` as value.

Reply:
What do you mean? Isn't it what you want?

Reply:
I means `undefined` valuated as a `string`.

Reply:
I still don't understand, I think that my implementation has pretty much the same behaviour as yours. Can you explain what's different?

Reply:
The difference is the `urlState` check. Please see the example bellow which describes this case

```js
const reg = /(.+)/
const [, result ] = reg.exec()
result === 'undefined' // true
```

Reply:
Oh ok, thanks. Because the `undefined` argument is cast into a string, that makes sense. So adding `''` as the default value for `urlState` should do it.

Reply:
Done :+1: